python/dgl/dataloading/dataloader.py

```
@@ -1,900 +1,602 @@
-            self.g_sampling, seed_nodes, exclude_eids=exclude_eids)
-
-        return input_nodes, pair_graph, neg_pair_graph, blocks
-
-    def collate(self, items):
-        """Combines the sampled edges into a minibatch for edge classification, edge
-        regression, and link prediction tasks.
-
-        Parameters
-        ----------
-        items : list[int] or list[tuple[str, int]]
-            Either a list of edge IDs (for homogeneous graphs), or a list of edge type-ID
-            pairs (for heterogeneous graphs).
-
-        Returns
-        -------
-        Either ``(input_nodes, pair_graph, blocks)``, or
-        ``(input_nodes, pair_graph, negative_pair_graph, blocks)`` if negative sampling is
-        enabled.
-
-        input_nodes : Tensor or dict[ntype, Tensor]
-            The input nodes necessary for computation in this minibatch.
-
-            If the original graph has multiple node types, return a dictionary of
-            node type names and node ID tensors.  Otherwise, return a single tensor.
-        pair_graph : DGLGraph
-            The graph that contains only the edges in the minibatch as well as their incident
-            nodes.
-
-            Note that the metagraph of this graph will be identical to that of the original
-            graph.
-        negative_pair_graph : DGLGraph
-            The graph that contains only the edges connecting the source and destination nodes
-            yielded from the given negative sampler, if negative sampling is enabled.
-
-            Note that the metagraph of this graph will be identical to that of the original
-            graph.
-        blocks : list[DGLGraph]
-            The list of MFGs necessary for computing the representation of the edges.
-        """
-        if self.negative_sampler is None:
-            return self._collate(items)
-        else:
-            return self._collate_with_negative_sampling(items)
+            dgl_warning(
```

Throw error instead of warning?